### PR TITLE
Feature/search within collection set

### PR DESCRIPTION
**Pull request by appsol**
### Summary
https://trello.com/c/R2OD8fxL

- Change search parameters persona and category to accept either a single collection slug, or a comma separated set of collection slugs.
- Expanded search tests to ensure query and location searches occur within the specified collections

### Development checklist
- [ ] Changes have been made to the API?
  - [ ] If so, the OpenAPI specification has been updated
- [ ] Database migrations have been added?
  - [ ] If so, the MySQL Workbench ERD has been updated
- [x] The code has been linted `./develop composer fix:style`

### Release checklist
Re-index Elasticsearch

### Notes
NA

### CODE REVIEW

File: app/Http/Controllers/Core/V1/SearchController.php [hunk 1]
```diff
@@ -61,6 +65,8 @@ public function __invoke(Search $search, Request $request)
         // Apply order.
         $search->applyOrder($request->order ?? 'relevance', $location ?? null);
 
+        // dd($search->getQuery());
+
```
Comment (matthew-inamdar) [inline app/Http/Controllers/Core/V1/SearchController.php:69]: Please remove.

```suggestion
        // dd($search->getQuery());

```

File: app/Search/ElasticsearchSearch.php [hunk 2]
```diff
@@ -151,9 +151,17 @@ public function applyCategory(string $category): Search
...
         }
 
+        foreach ($this->query['query']['bool']['filter']['bool']['must'] as &$filter) {
+            if (is_array($filter) && array_key_exists('terms', $filter) && array_key_exists('collection_categories', $filter['terms'])) {
+                $filter['terms']['collection_categories'][] = $categoryModel->name;
+
+                return $this;
+            }
+        }
+
```
Comment (matthew-inamdar) [inline app/Search/ElasticsearchSearch.php:161]: What's the purpose of this? Can we not only use the `terms` query you've done below?

File: app/Search/ElasticsearchSearch.php [hunk 3]
```diff
@@ -177,9 +185,17 @@ public function applyPersona(string $persona): Search
...
         }
 
+        foreach ($this->query['query']['bool']['filter']['bool']['must'] as &$filter) {
+            if (is_array($filter) && array_key_exists('terms', $filter) && array_key_exists('collection_personas', $filter['terms'])) {
+                $filter['terms']['collection_personas'][] = $categoryModel->name;
+
+                return $this;
+            }
+        }
+
```
Comment (matthew-inamdar) [inline app/Search/ElasticsearchSearch.php:195]: What's the purpose of this? Can we not only use the `terms` query you've done below?

**Review by matthew-inamdar — CHANGES_REQUESTED** [review]
Please see code comments.

OpenAPI spec updates are also needed.

File: app/Http/Controllers/Core/V1/SearchController.php (hunk 1 above)
Comment (appsol) [reply app/Http/Controllers/Core/V1/SearchController.php:69]: Sorry, my bad.

File: app/Search/ElasticsearchSearch.php (hunk 2 above)
Comment (appsol) [reply app/Search/ElasticsearchSearch.php:161]: Agreed both are very similar, but I was trying to follow the pattern that existed. I've combined it into one method that takes a slug and a type string to either apply a persona or a category filter.

File: app/Search/ElasticsearchSearch.php (hunk 3 above)
Comment (appsol) [reply app/Search/ElasticsearchSearch.php:195]: See above

**Comment by appsol** [comment]
@matthew-inamdar The OpenAPI specs have not changed, the search object still accepts a category or persona string, but these are now slugs, not names. Additionally they can now be either a single slug or a comma separated series of slugs, but either way it is still a string.

File: app/Contracts/Search.php [hunk 4]
```diff
@@ -24,15 +24,10 @@ public function applyType(string $type): Search;
...
      * @return \App\Contracts\Search
      */
-    public function applyCategory(string $category): Search;
-
-    /**
-     * @param string $persona
-     * @return \App\Contracts\Search
-     */
-    public function applyPersona(string $persona): Search;
+    public function applyCollection(string $category, string $type): Search;
```
Comment (matthew-inamdar) [inline app/Contracts/Search.php:30]: Can we revert this? I think the interface was more explicit and clearer before. Understanding the method implementations may have been almost identical, I'd prefer if we instead used a protected method in the implementation class which these methods could delegate to.

**Review by matthew-inamdar — CHANGES_REQUESTED** [review]
Comments left

File: app/Contracts/Search.php (hunk 4 above)
Comment (appsol) [reply app/Contracts/Search.php]: I've reverted to the previous api, but as you suggested have centralised the functionality in a protected method which both public methods delegate to.

**Review by matthew-inamdar — APPROVED** [review]

**Pull request merged**